weasyl/test/test_login.py

```diff
@@ -0,0 +1,598 @@
+    user_id = db_utils.create_user(username=user_name, password=raw_password)
+    d.engine.execute("UPDATE login SET settings = 'b' WHERE userid = %(id)s", id=user_id)
+    result = login.authenticate_bcrypt(username=user_name, password=raw_password)
+    assert result == (user_id, 'banned')
+
+def testAuthenticateBcrypt_LoginFailsForSuspendedUsersWithActiveDuration():
+    user_name = TestFunctions().generateTestAccountName()
+    user_id = db_utils.create_user(username=user_name, password=raw_password)
+    d.engine.execute("UPDATE login SET settings = 's' WHERE userid = %(id)s", id=user_id)
+    release_date = d.convert_unixdate(31, 12, 2030)
+    d.engine.execute("INSERT INTO suspension VALUES (%(id)s, %(reason)s, %(rel)s)",
+                     id=user_id, reason='test', rel=release_date)
+    result = login.authenticate_bcrypt(username=user_name, password=raw_password, session=False)
+    assert result == (user_id, 'suspended')
+
+def testAuthenticateBcrypt_LoginSucceedsForSuspendedUsersWithExpiredDuration():
+    user_name = TestFunctions().generateTestAccountName()
+    user_id = db_utils.create_user(username=user_name, password=raw_password)
+    d.engine.execute("UPDATE login SET settings = 's' WHERE userid = %(id)s", id=user_id)
+    release_date = d.convert_unixdate(31, 12, 2015)
+    d.engine.execute("INSERT INTO suspension VALUES (%(id)s, %(reason)s, %(rel)s)",
+                     id=user_id, reason='test', rel=release_date)
+    result = login.authenticate_bcrypt(username=user_name, password=raw_password, session=False)
+    assert result == (user_id, None)
+
+# For some reason getting the password to work with the accented E just isn't happening...
+#     def testLoginSucceedsForValidUserAndPasswordWithUnicodeFailure(self):
+#         # $2a$12$qReI924/8pAsoHu6aRTX2ejyujAZ/9FiOOtrjczBIwf8wqXAJ22N. == "passwprd"
+#         user_id = db_utils.create_user(username='testloginsuite')
+#         d.execute("INSERT INTO authbcrypt VALUES (%i, '%s')", [user_id, old_2a_bcrypt_hash])
+#         d.execute("UPDATE authbcrypt SET hashsum = '%s' WHERE userid = %i", [login.passhash(raw_password), user_id])
+#         result = login.authenticate_bcrypt(username='testloginsuite', password=u'passwordé', session=False)
+#         print result
+#         self.assertTrue(result[0] == user_id)
+#         self.assertTrue(result[1] == 'unicode-failure')
+#         d.execute("DELETE FROM login WHERE userid = %i", [user_id])
+
+def testAuthenticateBcrypt_LoginSucceedsForValidUserAndPassword():
+    user_name = TestFunctions().generateTestAccountName()
+    user_id = db_utils.create_user(username=user_name, password=raw_password)
+    result = login.authenticate_bcrypt(username=user_name, password=raw_password, session=False)
+    assert result == (user_id, None)
+
+"""
+Test section for functions:
+a) login.py::def password_secure(password):
+b) login.py::def passhash(password):
+"""
+def testPasswordSecure_VerfyFunctionality():
```

Comment:
camelCase with underscores doesn’t make for great names. I suggest calling this one `test_password_secure`.